### PR TITLE
Firewall Rules - visibility adjustments

### PR DESCRIPTION
**Pull request by commonism**
partially addressing #2583

### CODE REVIEW

File: src/www/firewall_rules.php [hunk 1]
```diff
@@ -367,8 +376,8 @@
                       <th class="hidden-xs hidden-sm"><?=gettext("Port");?></th>
                       <th class="hidden-xs hidden-sm"><?=gettext("Destination");?></th>
                       <th class="hidden-xs hidden-sm"><?=gettext("Port");?></th>
-                      <th class="hidden-xs hidden-sm"><?=gettext("Gateway");?></th>
-                      <th class="hidden-xs hidden-sm"><?=gettext("Schedule");?></th>
+                      <th class="hidden"><?=gettext("Gateway");?></th>
```
Comment (AdSchellevis) [inline src/www/firewall_rules.php]: I don't think we should hide gateway (and schedule) columns without them being visible on the overview page completely, policy based routing mistakes tend to cause quite some issues in peoples setups.

File: src/www/firewall_rules.php [hunk 2]
```diff
@@ -477,6 +489,28 @@
                         $selected_if == 'FloatingRules'
                      )
                 ):
+				  if( $selected_if != $interface )
+                  {
+					$offset = $i;
+                    $interface = $selected_if;
+                  }
+                  $idx = $i-$offset;
```
Comment (AdSchellevis) [inline src/www/firewall_rules.php]: I'm not sure what this is supposed to be doing, if we only want to add a number to the rule, this feels a bit too much.

File: src/www/firewall_rules.php [hunk 3]
```diff
@@ -477,6 +489,28 @@
...
+                  if ($filterent['category'] !== $category) {
+                    $category = $filterent['category']
+?>
+
+                  <tr class="category text-uppercase text-danger text-center font-weight-bold" data-category="<?=!empty($filterent['category']) ? $filterent['category'] : "";?>">
+                    <td colspan=2></td>
+                    <td colspan=3><?=$category?></td>
+                    <td></td>
+                    <td></td>
+                  </tr>
```
Comment (AdSchellevis) [inline src/www/firewall_rules.php]: Optically this doesn't look really nice on my end, I've attached a screenshot of what it looks like now
image

File: src/www/firewall_rules.php (hunk 1 above)
Comment (commonism) [reply src/www/firewall_rules.php]: Just trying to serve the 80% here.
A single column wich can be used to indicate 'special' configuration options like Gateway, Schedule, statetype, .... maybe?

File: src/www/firewall_rules.php (hunk 2 above)
Comment (commonism) [reply src/www/firewall_rules.php]: Numbering rules per interface helps addressing the *same* rule in a discussion.
I'm not aware of an easier way to get the number.
The rules data structure is not per interface, therefore I choose to store the interfaces offset.

File: src/www/firewall_rules.php (hunk 3 above)
Comment (commonism) [reply src/www/firewall_rules.php]: I think rendering issue is caused by an empty category for rule 3?

**Comment by fichtner** [comment]
this needs to be redone due to conflicts

**Pull request closed without merging**